imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/DefaultImageFileLoader.java

```diff
@@ -166,6 +166,15 @@ public void run() {
                                 folderMap.put(bucket, folder);
                             }
                             folder.getImages().add(image);
+                        }else if(bucket == null){
```

Comment:
I think we can simplify this and further null-proof the code by changing this to

```java

// this code should be placed before if (name != null)

if (bucket == null) {
  File parent = new File(path).getParentFile();
  if (parent != null) {
    bucket = parent.getName();
  } else {
    bucket = "SDCARD";
  }
}
```

Reply:
@esafirm Thanks for your effort, I have push it to this PR